### PR TITLE
Move testing modules to a testing directory.

### PR DESCRIPTION
**Pull request by jplaisted**
This is per discussion from the Elasticsearch Integration Testing RFC (#15).

#17 

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -29,7 +29,10 @@ project.ext.spec = [
     ]
 ]
 
+def junitJupiterVersion = '5.6.1'
```
Comment (mars-lan) [inline build.gradle:32]: Move to `buildscript.ext`?

File: dao-api/build.gradle [hunk 2]
```diff
@@ -15,8 +15,8 @@ dependencies {
 
   annotationProcessor externalDependency.lombok
 
-  testCompile project(':test-models')
-  testCompile project(':core-models-testing')
+  testCompile project(':testing:core-models-testing')
```
Comment (mars-lan) [inline dao-api/build.gradle:18]: swap L18 & 19?

**Review by mars-lan — COMMENTED** [review]
Otherwise LGTM, assuming CI passes.

File: build.gradle (hunk 1 above)
Comment (jplaisted) [reply build.gradle:32]: Reverted

**Review by mars-lan — APPROVED** [review]

**Pull request merged**